cocotb/share/lib/py_gpi_log/py_gpi_logging.cpp

```diff
@@ -129,8 +141,8 @@ static void py_gpi_log_handler(void *, const char *name, int level,
 error:
     // LCOV_EXCL_START
     /* Note: don't call the LOG_ERROR macro because that might recurse */
-    gpi_native_logger_vlog(name, level, pathname, funcname, lineno, msg,
-                           argp_copy);
+    gpi_native_logger_log(name, level, pathname, funcname, lineno,
+                          log_buff.data());
```

Comment:
The `goto error`s at the top of the function will skip updating `log_buff`, so it would print whatever the last log message was.

Reply:
I can move all the Python checks to below the formatting. The message needs to be formatted and put into the buffer regardless, and if the formatting fails we don't want to try to print the message anyways. Maybe, I'll just rewrite this whole function and use the `defer` from gpi_embed.cpp to make my life a little easier. `goto error` sucks.